standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -4259,6 +4261,15 @@ public void scheduledQueryProgress(ScheduledQueryProgressInfo info) throws TExce
     client.scheduled_query_progress(info);
   }
 
+  @Override
+  public void setValidWriteIdList(String txnWriteIdListStr) {
+    this.txnWriteIdList = (txnWriteIdListStr == null ? null : new ValidTxnWriteIdList(txnWriteIdListStr));
+  }
+
+  @Override public void clearValidWriteIdList() {
+    this.txnWriteIdList = null;
+  }
+
```

Comment:
If I understand correctly this is also only relevant for the optimization. Could we also put this into another jira?

Reply:
I did create separate subtasks, but after internal discussions, I decided to club the changes, as they are closely related and work together to provide cache consistency.